lookout/style/format/annotations/annotated_data.py

```diff
@@ -0,0 +1,329 @@
+            self._add(annotation)
+
+    def _add(self, annotation: Annotation) -> None:
+        """
+        Add an annotation. Annotations of the same type may not overlap.
+        """
+        annotation_type = type(annotation)
+        if annotation.start == 0 and annotation.stop == len(self):
+            if annotation_type in self._global_annotations:
+                raise ValueError("Global annotation %s already exists" % annotation)
+            self._global_annotations[annotation_type] = annotation
+        else:
+            if annotation.span not in self._span_to_annotations:
+                self._span_to_annotations[annotation.span] = {}
+            if annotation_type not in self._type_to_annotations:
+                self._type_to_annotations[annotation_type] = SortedDict()
+            if self._type_to_annotations[annotation_type]:
+                right_span_index = self._type_to_annotations[annotation_type].bisect_left(
+                    annotation.span)
+                if right_span_index < len(self._type_to_annotations[annotation_type]):
+                    right_span, right_annotation = \
+                        self._type_to_annotations[annotation_type].peekitem(right_span_index)
+                    if self._check_spans_overlap(*right_span, *annotation.span):
+                        raise ValueError(
+                            "Trying to add Annotation %s which overlaps with existing %s" % (
+                                right_annotation, annotation))
+                left_span_index = max(0, right_span_index - 1)
+                left_span, left_annotation = self._type_to_annotations[annotation_type].peekitem(
+                    left_span_index)
+                if self._check_spans_overlap(*left_span, *annotation.span):
+                    raise ValueError("Trying to add Annotation %s which overlaps with existing"
+                                     "%s" % (left_annotation, annotation))
+            self._span_to_annotations[annotation.span][annotation_type] = annotation
+            self._type_to_annotations[annotation_type][annotation.span] = annotation
+
+    def get(self, annotation_type: Type[Annotation], span: Optional[Tuple[int, int]] = None,
+            ) -> Annotation:
+        """
+        Return an annotation for the given span and type.
+
+        Looking for an exact (type and span) match only.
+
+        :param annotation_type: Annotation type to get.
+        :param span: Annotation span (range) to get. If span is not specified it returns an \
+                     annotation that cover all content (aka global annotation).
+        :return: Requested `Annotation`.
+        """
+        if span is None:
+            return self._global_annotations[annotation_type]
+        check_span(*span)
+        return self._type_to_annotations[annotation_type][span]
+
+    def iter_by_type_nested(self, annotation_type: Type[Annotation],
+                            *additional: Type[Annotation],
```

Comment:
```suggestion
                            *covered_by: Type[Annotation],
```

Because "additional" does not reflect the whole meaning.